tests/test_multistore_file.py

```diff
@@ -41,3 +74,170 @@ def test_lock_file_raises_ioerror(self):
                 self.assertTrue(multistore._file.open_and_lock_called)
         finally:
             os.unlink(filename)
+
+    def test_read_only_file_fail_lock(self):
+        credentials = self.create_test_credentials()
+
+        open(FILENAME, 'a+b').close()
+        os.chmod(FILENAME, 0o400)
+
+        store = multistore_file.get_credential_storage(
+            FILENAME,
+            credentials.client_id,
+            credentials.user_agent,
+            ['some-scope', 'some-other-scope'])
+
+        store.put(credentials)
+        if os.name == 'posix':
```

Comment:
Haha. You can always create a Windows VM on Compute to check. (Not necessary now, this is someone else's legacy.)